src/Data/NonEmpty.purs

```diff
@@ -47,8 +49,8 @@ singleton :: forall f a. Plus f => a -> NonEmpty f a
 singleton a = a :| empty
 
 -- | Fold a non-empty structure, collecting results using a binary operation.
-foldl1 :: forall f a. Foldable f => (a -> a -> a) -> NonEmpty f a -> a
-foldl1 f (a :| fa) = foldl f a fa
+foldl1 :: forall f a. Foldable f => Warn (Text "'foldl1' is deprecated") => (a -> a -> a) -> NonEmpty f a -> a
```

Comment:
Should we give a suggestion there?

```
'Data.NonEmpty.foldl1' is deprecated, use  'Data.Semigroup.Foldable.foldl1' instead
```

Reply:
I think that's a good idea -- better to suggest the proper alternative as there is one.